File: TiaCodegen/Interfaces/IOperationOrSignalDirectionWrapper.cs
using TiaCodegen.Enums;

namespace TiaCodegen.Interfaces
{
    public class IOperationOrSignalDirectionWrapper
    {
        public IOperationOrSignal OperationOrSignal { get; set; }

        public Direction Direction { get; set; }

        public int Length { get; set; }

        public SignalType? Type { get; set; }

        public IOperationOrSignalDirectionWrapper(IOperationOrSignal operationOrSignal, Direction direction, SignalType? type = null, int length = 0)
        {
            OperationOrSignal = operationOrSignal;
            Direction = direction;
            Length = length;
            Type = type;
        }
    }
}

File: TiaCodegen-ts/src/Interfaces/IOperationOrSignalDirectionWrapper.ts
import { IOperationOrSignal } from './IOperationOrSignal';
import { Direction } from '../Enums/Direction';
import { SignalType } from '../Enums/SignalType';

export class IOperationOrSignalDirectionWrapper {
    operationOrSignal: IOperationOrSignal | null;
    direction: Direction;
    length: number;
    type: SignalType | null;

    constructor(
        operationOrSignal: IOperationOrSignal | null,
        direction: Direction,
        type: SignalType | null = null,
        length: number = 0,
    ) {
        this.operationOrSignal = operationOrSignal;
        this.direction = direction;
        this.length = length;
        this.type = type;
    }
}
